simulator/risc_v/riscv_register/riscv_register.cpp

```diff
@@ -36,3 +36,21 @@ RISCVRegister::RegNum RISCVRegister::get_csr_regnum( std::string_view name)
 #undef DECLARE_CSR
     return MAX_VAL_RegNum;
 }
+
+RISCVRegister RISCVRegister::from_cpu_index( uint8 id, RISCVRegister::separator type)
```

Comment:
This is an external interface and it makes no sense to provide an additional parameter here.

Reply:
For generating a RISCVRegister for "popular" registers, consider a new function:
```c++
static RISCVRegsiter from_popular_index( uint8 id)
```